tldr depends on curl

I'm getting a `src/net.c:4:23: fatal error: curl/curl.h: No such file or directory` error during the build.
 
Here are the relevant logs: https://gist.github.com/anonymous/3635bed06a7c741e54ee3e01513f578c

/cc @Leandros, @danielcompton, @zmwangx and @DomT4 per #4637. Particularly, [this exchange](https://github.com/Homebrew/homebrew-core/pull/4637#discussion-diff-78221277) seems relevant.

Here's the original output, btw:

```
$ brew install tldr
Updating Homebrew...
==> Installing dependencies for tldr: patchelf, pkg-config, libzip
==> Installing tldr dependency: patchelf
==> Downloading https://linuxbrew.bintray.com/bottles/patchelf-0.9_1.x86_64_linux.bottle.tar.gz
######################################################################## 100,0%
==> Pouring patchelf-0.9_1.x86_64_linux.bottle.tar.gz
🍺  /home/waldyrious/.linuxbrew/Cellar/patchelf/0.9_1: 6 files, 1.2M
==> Installing tldr dependency: pkg-config
==> Downloading https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.1.tar.gz
######################################################################## 100,0%
==> ./configure --prefix=/home/waldyrious/.linuxbrew/Cellar/pkg-config/0.29.1_2 --disable-host-tool --with-internal-glib --with-pc-path=/home/waldyrious/.linuxbrew/lib/pkgconfig:/home/waldyrious/.linuxbrew/share/pkgconfig:/usr/local/lib/pk
==> make
==> make check
==> make install
🍺  /home/waldyrious/.linuxbrew/Cellar/pkg-config/0.29.1_2: 11 files, 677.8K, built in 57 seconds
==> Installing tldr dependency: libzip
==> Downloading https://linuxbrew.bintray.com/bottles/libzip-1.1.2.x86_64_linux.bottle.tar.gz
######################################################################## 100,0%
==> Pouring libzip-1.1.2.x86_64_linux.bottle.tar.gz
🍺  /home/waldyrious/.linuxbrew/Cellar/libzip/1.1.2: 120 files, 617.9K
==> Installing tldr 
==> Downloading https://github.com/tldr-pages/tldr-cpp-client/archive/v1.3.0.tar.gz
==> Downloading from https://codeload.github.com/tldr-pages/tldr-cpp-client/tar.gz/v1.3.0
######################################################################## 100,0%
==> make PREFIX=/home/waldyrious/.linuxbrew/Cellar/tldr/1.3.0 install
Last 15 lines from /home/waldyrious/.cache/Homebrew/Logs/tldr/01.make:

make
PREFIX=/home/waldyrious/.linuxbrew/Cellar/tldr/1.3.0
install

CC src/local.c
CC src/tldr.c
CC src/net.c
CC src/parser.c
CC src/utils.c
src/net.c:4:23: fatal error: curl/curl.h: No such file or directory
compilation terminated.
Makefile:119: recipe for target 'obj/net.o' failed
make: *** [obj/net.o] Error 1
```

--- a/Formula/tldr.rb
+++ b/Formula/tldr.rb
@@ -15,6 +15,7 @@ class Tldr < Formula
 
   depends_on "pkg-config" => :build
   depends_on "libzip"
+  depends_on "curl" unless OS.mac?
 
   def install
     system "make", "PREFIX=#{prefix}", "install"

